src/software/amazon/ion/IntegerType.java

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2016 Amazon.com, Inc.  All rights reserved.
+
+package software.amazon.ion;
+
+/**
+ * Indicates the Java type of an Ion <code>int</code> value.
+ */
+public enum IntegerType
```

Comment:
The Ion type is called `int`, so why isn't this `IntType`?

I'm also not convinced that "type" it the best term here.  The type (in Ion's type system) is `int`. This is a different kind of classification within that type.

Oh wait... I see this is the _Java_ type being discussed.  So there's ambiguity throughout this feature.  Perhaps it could be reduced by talking about "size" rather than "type"?